lib/utils/exploits.ts

```diff
@@ -81,12 +83,44 @@ export const getUniqueChainIds = (exploit: Exploit): number[] => {
   return sortedChainIds;
 };
 
-export const isAffectedByExploit = (allowances: AllowanceData[], exploit: Exploit) => {
-  return allowances.some((allowance) => {
-    return exploit.addresses.some((address) => {
-      return address.chainId === allowance.chainId && address.address === allowance.spender;
+export const getExploitStatus = (allowances: AllowanceData[], exploit: Exploit) => {
+  const { events } = useAddressEvents();
+  const parseEvents = (logs: Log[]) => logs.map(parseLog).filter((event): event is ParsedEvent => event !== null);
+
+  const parsedEvents = [
+    ...parseEvents(events.approval),
+    ...parseEvents(events.approvalForAll),
+    ...parseEvents(events.permit2Approval),
+  ];
+
+  const isAddressAffected = (parsedEvent: ParsedEvent, exploitAddress: { chainId: number; address: string }) => {
+    return (
+      parsedEvent.owner === exploitAddress.address &&
+      exploitAddress.chainId === allowances.find((a) => a.spender === parsedEvent.spender)?.chainId
+    );
+  };
+
+  const affected = allowances.some((allowance) => {
+    return exploit.addresses.some((exploitAddress) => {
+      return (
+        exploitAddress.chainId === allowance.chainId &&
+        exploitAddress.address === allowance.spender &&
+        allowance.amount > 0n
```

Comment:
We can remove this `amount` check since only *active* allowances are included in the `allowances`.

Reply:
oo my bad, thought I removded